### PR TITLE
#1214: field grouping documentation.

### LINKED ISSUE #1214: Add documentation for JSON field groups
- Original PR introducing the feature: [#410](https://github.com/area17/twill/pull/410)
- Recent improvements with casts and translations: [#1151](https://github.com/area17/twill/pull/1151)

### PR DESCRIPTION
**Pull request by haringsrob**
## Description

Still working on this as there seems to be a bug with translatable json fields.

This pr also fixes translatable attributes in json.

## Related Issues

fixes #1214

### CODE REVIEW

**Comment by ifox** [comment]
Looks great, thank you! Was https://github.com/area17/twill/pull/1151 incomplete regarding translated fields?

File: src/Models/Model.php [hunk 1]
```diff
@@ -109,6 +109,10 @@ protected static function bootTaggableTrait()
         static::$tagsModel = Tag::class;
     }
 
+    public function implementsTrait(string $trait): bool  {
```
Comment (pboivin) [inline src/Models/Model.php:112]: Could this be used instead? https://github.com/area17/twill/blob/2.x/src/Helpers/helpers.php#L49

**Comment by pboivin** [comment]
@haringsrob The added documentation looks great! I'm not sure I follow re. what's being added in `getFormFieldsHandleFieldsGroups()`. I'll do a quick test on my end to refresh my memory on this.

**Comment by haringsrob** [comment]
Well, what is happening is that even if you have a json field translatable it would return a flat array.

Per the docs the $fields to the frontend would contain:

```
link_target = target
link_url = url
link_label = label
```

This wont work as the fields are marked as translatable.
So the front-end expects this:

```
link_target = en: target, nl: target
link_url = en:URL, nl:url
link_label = en:label, nl: label
```

**Comment by pboivin** [comment]
Does this have anything to do with the `cast` property on the model? IIRC this is needed on the translated model.

**Comment by haringsrob** [comment]
With these changes this is the result in the db:

**Comment by pboivin** [comment]
@haringsrob The following configuration works for me on https://github.com/area17/twill/pull/1151

Does this make any sense?

#### Migration

```
Schema::create('page_translations', function (Blueprint $table) {
    createDefaultTranslationsTableFields($table, 'page');
    $table->string('title', 200)->nullable();
    $table->text('description')->nullable();
    $table->json('footer')->nullable();
});
```

#### Model

```
protected $fillable = [
    'footer',
];

public $translatedAttributes = [
    'footer',
];
```

#### Translation Model

```
public $casts = [
    'footer' => 'array',
];
```

#### Repository

```
protected $fieldsGroups = [
    'footer' => ['title', 'subtitle'],
];

public $fieldsGroupsFormFieldNamesAutoPrefix = true;
```

#### Form

```
@formField('input', [
    'name' => 'footer_title',
    'label' => 'footer Title',
    'translated' => true,
    'maxlength' => 100
])

@formField('input', [
    'name' => 'footer_subtitle',
    'label' => 'footer Subtitle',
    'translated' => true,
    'maxlength' => 100
])
```

#### Frontend

```
{{ $page->footer['title'] }}
{{ $page->footer['subtitle'] }}
```

File: src/Models/Model.php (hunk 1 above)
Comment (haringsrob) [reply src/Models/Model.php:112]: The class uses recursive is a laravel function that does the same.

https://laravel.com/docs/8.x/helpers#method-class-uses-recursive

But this is no longer needed as my change were not required :D

**Pull request merged**